multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java

```diff
@@ -6,96 +6,72 @@
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
 
+@Slf4j
 @Service
 public class StorageManager {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(StorageManager.class);
   private static final String INDEXED_PRODUCT_METADATA_CALLBACK_TYPE = "cst";
 
-  private final Dao<Product, String> productDao;
-  private final Dao<IndexedProductMetadata, String> cstDao;
-  private final DataRetriever retriever;
+  @NotEmpty private final String retrieveEndpoint;
+
+  @NotNull private final S3StorageAdaptor s3;
+  @NotNull private final Dao<IndexedProductMetadata, String> cstDao;
 
-  @Autowired
   public StorageManager(
-      Dao<Product, String> productDao,
-      Dao<IndexedProductMetadata, String> cstDao,
-      DataRetriever retriever) {
-    this.productDao = productDao;
+      @NotEmpty @Value("${endpointUrl.retrieve}") final String retrieveEndpoint,
+      @NotNull final S3StorageAdaptor s3,
+      @NotNull final Dao<IndexedProductMetadata, String> cstDao) {
+    this.retrieveEndpoint = retrieveEndpoint;
+    this.s3 = s3;
     this.cstDao = cstDao;
-    this.retriever = retriever;
   }
 
-  /**
-   * @throws RetrievalClientException if there is a 400 status code when sending the request from
-   *     the callback
-   * @throws RetrievalServerException if there was a 500 status code when sending the request from
-   *     the callback
-   * @throws com.connexta.multiintstore.services.api.DuplicateIdException if there was an Id that
-   *     already exists
-   * @throws com.connexta.multiintstore.services.api.StorageException if there was an error storing
-   *     the callback
-   */
-  public void handleGeneralCallback(Object callback, String ingestId)
-      throws StorageException, DuplicateIdException, RetrievalServerException,
-          RetrievalClientException {
+  public URL storeProduct(
+      String acceptVersion, Long fileSize, String mimeType, MultipartFile file, String fileName)
+      throws IOException, StorageException {
 
-    if (callback instanceof ProductCallback) {
-      handleCallback((ProductCallback) callback);
-    }
-    if (callback instanceof FinishedCallback) {
-      handleCallback((FinishedCallback) callback);
-    }
-    if (callback instanceof MetadataCallback) {
-      handleCallback((MetadataCallback) callback, ingestId);
-    }
-  }
+    // TODO: Validate Accept-Version
+    final String key = UUID.randomUUID().toString().replace("-", "");
 
-  private void handleCallback(ProductCallback callback) {
-    LOGGER.info("Product");
-    //  TODO :: Check Markings
-    //  TODO :: Store Product
+    // Store in S3
+    s3.store(mimeType, file, fileSize, fileName, key);
+    // return product location
+    return new URL(retrieveEndpoint + key);
```

Comment:
✏️ How should failing to construct the URL be handled?

Reply:
Probably just the same way we do it in other places: catch a MalformedURLException

Reply:
Actually, we already throw an IOException and the caller logs the message. I think we're ok leaving it as is.